backends/bmv2/common/expression.cpp

```diff
@@ -251,7 +251,9 @@ void ExpressionConverter::postorder(const IR::Member* expression)  {
         result->emplace("type", "hexstr");
         auto decl = type->to<IR::Type_Error>()->getDeclByName(expression->member.name);
         auto errorValue = structure->errorCodesMap.at(decl);
-        result->emplace("value", Util::toString(errorValue));
+        // this generates error constant like hex value
+        auto reprValue = stringRepr(errorValue);
```

Comment:
if this is a bug fix you should probably file a separate PR for it, including a test case.

Reply:
Thank you for all comments.
I think that it is a bug fix, because when I extended core.p4 errors, I had noticed  that two tests had failed: checksum3-bmv2.p4 and checksum2-bmv2.p4. These tests reported out_of_range exception when make check is done, because tests added new errors which will have values 9 and 10, but it didn't generate error values as hex values (0x9 and 0xa) in json file. For example, in case of error value 10 it was generated like:
    "type" : "hexstr",
     "value" : "10"
which will be read as hex value 16, not 10 as it should be.
Now it is generate like:
  "type" : "hexstr",
   "value" : "0xa"
I will include a test for this case and make it as a seperate PR.